cpg_workflows/jobs/mito.py

```diff
@@ -0,0 +1,327 @@
+
+    Args:
+        metrics: CollectWgsMetrics metrics output to process.
+        mean_path: Output path for mean file.
+        median_path: Output path for median file.
+
+    Outputs:
+        job.mean_coverage: mean coverage of chrM
+        job.median_coverage: median coverage of chrM
+    """
+    job_attrs = job_attrs or {}
+    j = b.new_job('extract_coverage_mean', job_attrs)
+    j.image(image_path('peer'))
+
+    res = STANDARD.request_resources(ncpu=2)
+    res.set_to_job(j)
+
+    cmd = f"""
+
+    R --vanilla <<CODE
+        df = read.table(
+            "{metrics}",skip=6,header=TRUE,stringsAsFactors=FALSE,sep='\\\\t',nrows=1
+        )
+        write.table(
+            floor(df[,"MEAN_COVERAGE"]),
+            "{j.mean_coverage}",
+            quote=F, col.names=F, row.names=F)
+        write.table(
+            df[,"MEDIAN_COVERAGE"],
+            "{j.median_coverage}",
+            quote=F, col.names=F, row.names=F)
+        CODE
+    """
+
+    j.command(command(cmd))
+    if mean_path:
+        b.write_output(j.mean_coverage, str(mean_path))
+    if median_path:
+        b.write_output(j.median_coverage, str(median_path))
+
+    return j
+
+
+def coverage_at_every_base(
+    b,
+    cram: hb.ResourceGroup,
+    reference: hb.ResourceGroup,
+    intervals_list: hb.ResourceFile,
+    job_attrs: dict | None = None,
+) -> Job:
+    """
+    Run picard CollectHsMetrics to calculate read coverage at each base.
+
+    Args:
+        cram: Input cram
+        reference: Cram reference genome
+        intervals_list: intervals list of target region
+
+    Outputs:
+        job.per_base_coverage
+        job.hs_metrics_out
+    """
+    job_attrs = job_attrs or {}
+    j = b.new_job('coverage_at_every_base', job_attrs)
+    j.image(image_path('picard'))
+
+    res = STANDARD.request_resources(ncpu=2)
+    res.set_to_job(j)
+
+    cmd = f"""
+    picard CollectHsMetrics \
+      I={cram.cram} \
+      R={reference.base} \
+      PER_BASE_COVERAGE={j.per_base_coverage} \
+      O={j.hs_metrics_out} \
+      TI={intervals_list} \
+      BI={intervals_list} \
+      COVMAX=20000 \
+      SAMPLE_SIZE=1
+
+    """
+
+    j.command(command(cmd))
+
+    return j
+
+
+def merge_coverage(
+    b,
+    non_cr_coverage: hb.ResourceFile,
+    shifted_cr_coverage: hb.ResourceFile,
+    merged_coverage: Path,
+    job_attrs: dict | None = None,
+) -> Job:
+    """
+    Merge per base coverage files from non-control region and shifted control region.
+
+    Args:
+        non_cr_coverage: Per-base coverage from CollectHsMetrics for all of
+            chrM excluding the control region.
+        shifted_cr_coverage: Per-base coverage from CollectHsMetrics for only the chrM
+            control region in shifted coord space.
+        merged_coverage: Path to write merged coverage tsv.
+
+    Outputs:
+        job.merged_coverage: Merged coverage tsv.
+    """
+    job_attrs = job_attrs or {}
```

Comment:
ah there was only one more.
as above 😂

Reply:
Thanks @vivbak 😃  Job_attrs  are now populated across the stack and everything else cleaned up.